src/pybullet_industrial/gcode_class.py

```diff
@@ -0,0 +1,455 @@
+
+                position_error = np.linalg.norm(current_position
+                                                - self.new_point)
+
+                orientation_error = np.linalg.norm(current_orientation
+                                                   - self.new_or)
+
+                if position_error < 0.02 and orientation_error < 0.004:
+                    break
+        else:
+
+            for _ in range(20):
+                self.robot.set_endeffector_pose(self.new_point, orientation)
+                for _ in range(10):
+                    p.stepSimulation()
+                    time.sleep(self.sleep)
+
+                current_position = self.robot.get_endeffector_pose()[0]
+                or_euler = p.getEulerFromQuaternion(
+                    self.robot.get_endeffector_pose()[1])
+                current_orientation = np.array(or_euler)
+
+                position_error = np.linalg.norm(current_position -
+                                                self.new_point)
+                orientation_error = np.linalg.norm(current_orientation -
+                                                   self.new_or)
+
+                if position_error < 0.02 and orientation_error < 0.004:
+                    break
+
+    def g123_interpolation(self, g_com, r_val):
+        """This Mehtod is part of the run_gcode() Method. Depending on the
+        G-Command it either runs a linear or circular interpolation and calls
+        the move-methods depending if a tool is activated or not.
+
+        Args:
+            g_com: G-Command from the g_code File
+            r_val: Radius for the circular interpolation
+
+        Returns:
+            None
+        """
+        orientation = p.getQuaternionFromEuler(self.new_or)
+
+        # Building the Path if there is a linear G1 interpolation
+        if g_com == 1:
+            path = pi.linear_interpolation(self.last_point,
+                                           self.new_point,
+                                           self.interpolation_steps)
+
+        # Building the path if there is a circular interpolation
+        elif g_com in [2, 3]:
+            if g_com == 2:
+                path = pi.circular_interpolation(self.last_point,
+                                                 self.new_point, r_val,
+                                                 self.interpolation_steps,
+                                                 self.plane, True)
+            else:
+                path = pi.circular_interpolation(self.last_point,
+                                                 self.new_point, r_val,
+                                                 self.interpolation_steps,
+                                                 self.plane, False)
+
+        path.orientations = np.transpose([orientation]
+                                         * len(path.orientations[0]))
+
+        # Moving endeffector or the robot
+        if not self.active_endeffector == -1:
```

Comment:
both of these functions are almost identical. this is dangerous as bugs in both could only be fixed in one. Avoid the duplicates compare with the g0 function which has the same problem.

Better yet fold them into one such as in g0
Also the naming is very inconsistent it is not clear that move_along_path does the same as move_robot just with the endeffector!